blueprint/core/blueprint_ShadowView.h

```diff
@@ -89,6 +91,54 @@ namespace blueprint
         }
     };
 
+    template <typename Setter, typename ...Args>
+    const auto getYogaNodeFloatSetter(Setter setter, Args... args) {
+      return [=](const juce::var& value, YGNodeRef node) {
+        if(value.isDouble()) {
+          setter(node, args..., (float) value);
+          return true;
+        }
+        return false;
+      };
+    }
+
+    template <typename Setter, typename SetterPercent, typename ...Args>
+    const auto getYogaNodeDimensionSetter(Setter setter, SetterPercent setterPercent, Args... args) {
+      return [=, floatSetter = getYogaNodeFloatSetter(setter, args...)](const juce::var& value, YGNodeRef node) {
+        if (floatSetter(value, node))
+          return true;
+        if (value.isString() && value.toString().trim().contains("%"))
+        {
+          juce::String strVal = value.toString().retainCharacters("-1234567890.");
+          setterPercent(node, args..., strVal.getFloatValue());
+          return true;
+        }
+        setter(node, args..., YGUndefined);
+        return true;
+      };
+    }
+
+    template <typename Setter, typename SetterPercent, typename SetterAuto, typename ...Args>
+    const auto getYogaNodeDimensionAutoSetter(Setter setter, SetterPercent setterPercent, SetterAuto setterAuto, Args... args) {
+      return [=, nonAutoSetter = getYogaNodeDimensionSetter(setter, setterPercent, args...)](const juce::var& value, YGNodeRef node) {
+        if (value.isString() && value.toString() == "auto") {
+          setterAuto(node, args...);
+          return true;
+        }
+        return nonAutoSetter(value, node);
+      };
+    }
+
+    template <typename Setter, typename EnumMap>
+    const auto getYogaNodeEnumSetter(Setter setter, EnumMap &map) {
+      return [=](const juce::var& value, YGNodeRef node) {                       \
+        try {
+          setter(node, map.at(value));
+        } catch(const std::out_of_range& e) { /* TODO log something */ }
```

Comment:
Similar here, I'm thinking we should throw from these helper setters

Reply:
@nick-thompson Yes, I'll get this one in for sure